app/models/public_json.rb

```diff
@@ -8,4 +8,19 @@ def data=(data)
   def data
     ActiveSupport::Gzip.decompress(super)
   end
+
+  def cocina_hash
+    data && JSON.parse(data)
+  end
+
+  # The purpose here is to tell DSA which files we currently have (across all versions)
+  def files_by_md5
+    file_sets = cocina_hash.dig('structural', 'contains')
```

Comment:
It's not clear what this method returns if the `JSON.parse(data)` branch is not reached due to `data` being false-y. But in that case, does the thing returned respond to `#dig` (line 18)?

Reply:
I don't think this ever happens:
```
Purl.where(public_json: nil).count
 => 0
 ```

Reply:
If it's never `nil`, do we need the `data && ` part on line 13? Or are you saying that when that is false-y, `#files_by_md5` is never invoked?

Reply:
We no longer have that thanks to #816

Reply:
Excellent 🍻